tests/ui/js/ui-storyEditorTests.js

```diff
@@ -146,30 +153,113 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
                     listener: "sjrk.storyTelling.testUtils.verifyBlocksRemoved",
                     args: ["{storyEditor}.blockManager", 2]
                 }]
             },
+            {
+                name: "Test block reorderer",
+                expect: 12,
+                sequence: [{
+                    // add a first block
+                    jQueryTrigger: "click",
+                    element: "{storyEditor}.dom.storyAddTextBlock"
+                },
+                {
+                    event: "{storyEditor}.blockManager.events.viewComponentRegisteredWithManager",
+                    listener: "sjrk.storyTelling.testUtils.verifyBlockAdded",
+                    args: ["{storyEditor}.blockManager", "{arguments}.0", "sjrk.storyTelling.blockUi.editor.textBlockEditor"]
+                },
+                {
+                    // add a second block
+                    jQueryTrigger: "click",
+                    element: "{storyEditor}.dom.storyAddTextBlock"
+                },
+                {
+                    event: "{storyEditor}.blockManager.events.viewComponentRegisteredWithManager",
+                    listener: "sjrk.storyTelling.testUtils.verifyBlockAdded",
+                    args: ["{storyEditor}.blockManager", "{arguments}.0", "sjrk.storyTelling.blockUi.editor.textBlockEditor"]
+                },
+                {
+                    // force the reorderer to refresh so it knows about these blocks
+                    func: "{storyEditor}.reorderer.refresh"
+                },
+                {
+                    event: "{storyEditor}.reorderer.events.onRefresh",
+                    listener: "jqUnit.assert",
+                    args: ["Reorderer onRefresh fired as expected"]
+                },
+                {
+                    // Click its reorder button up, listen for reorderer onMove
+                    jQueryTrigger: "click",
+                    element: "{storyEditor}.model.lastAddedBlock.dom.moveBlockUpButton"
+                },
+                {
+                    event: "{storyEditor}.reorderer.events.onMove",
+                    listener: "jqUnit.assert",
+                    args: ["Reorderer onMove fired on moveBlockUp button clicked"]
+                },
+                {
+                    // Click its reorder button down, listen for reorderer onMove
+                    jQueryTrigger: "click",
+                    element: "{storyEditor}.model.lastAddedBlock.dom.moveBlockDownButton"
+                },
+                {
+                    event: "{storyEditor}.reorderer.events.onMove",
+                    listener: "jqUnit.assert",
+                    args: ["Reorderer onMove fired on moveBlockDown button clicked"]
+                },
+                {
+                    // Call reorderBlockUp, listen for reorderer onMove
+                    func: "{storyEditor}.reorderer.reorderBlockUp",
```

Comment:
The call to the `refresh` invoker was to force a refresh after blocks are added and to ensure the invoker was operating properly. I agree that, as an integration test, it's redundant. What I propose instead is to move the jQueryTrigger of `focusin` to right after the blocks are added, that way the sequence more accurately simulates the user's interaction with the editor.

Because the blocks are added dynamically, we can't get away without refreshing the reorderer before clicking on the "down" button.